net/node/infoUpdate.go

```diff
@@ -123,57 +123,40 @@ func getNodeAddr(n *node) NodeAddr {
 }
 
 func (node *node) reconnect(peer *node) error {
-	isTls := config.Parameters.IsTLS
+
 	addr := getNodeAddr(peer)
 	var ip net.IP
 	ip = addr.IpAddr[:]
 	nodeAddr := ip.To16().String() + ":" + strconv.Itoa(int(addr.Port))
 	log.Info("try to reconnect peer, peer addr is ", nodeAddr)
-	var conn net.Conn
-	var err error
-	if isTls {
-		conn, err = TLSDial(nodeAddr)
-		if err != nil {
-			return nil
-		}
-	} else {
-		conn, err = NonTLSDial(nodeAddr)
-		if err != nil {
-			return nil
-		}
-	}
-	t := time.Now()
-	peer.UpdateRXTime(t)
-	peer.tryTimes = 0
-	peer.conn = conn
-	peer.addr, err = parseIPaddr(conn.RemoteAddr().String())
-	peer.local = node
-
-	log.Info(fmt.Sprintf("Reconnect node %s connect with %s with %s",
-		conn.LocalAddr().String(), conn.RemoteAddr().String(),
-		conn.RemoteAddr().Network()))
-	go peer.rx()
-
-	if node.GetID() > peer.GetID() {
-		peer.SetState(HAND)
-		buf, _ := NewVersion(node)
-		peer.Tx(buf)
+
+	rand.Seed(time.Now().UnixNano())
+
+	<-time.After(time.Duration(rand.Intn(CONNMAXBACK)) * time.Second)
+	log.Debug("Back off time`s up, start connect node")
+	err := node.Connect(nodeAddr)
+
+	if err == nil {
+		node.RemoveNodeFrmNBLst(peer.GetID())
```

Comment:
The node need be reconnected should be removed in advance to avoid race the new neighbor node